sources/imperative/reader/ReaderAPI.swift

```diff
@@ -71,6 +71,46 @@ extension CSVReader {
             try self.init(configuration: configuration, buffer: buffer, decoder: decoder)
         }
     }
+
+    public enum Stream {
+        case stdin
+    }
+
+    /// Creates a reader instance that will be used to parse the given CSV stream.
+    ///
+    /// If the configuration's encoding hasn't been set and the input data doesn't contain a Byte Order Marker (BOM), UTF8 is presumed.
+    /// - parameter input: The stream type to be parsed.
+    /// - parameter configuration: Recipe detailing how to parse the CSV data (i.e. encoding, delimiters, etc.).
+    /// - throws: `CSVError<CSVReader>` exclusively.
+    public convenience init(input: Stream, configuration: Configuration = .init()) throws {
+        if configuration.presample {
+            // A. If the `presample` configuration has been set, read all data from STDIN.
+            try self.init(input: FileHandle.standardInput.readDataToEndOfFile(), configuration: configuration); return
+        } else {
+            // B. Otherwise, create an input stream and start parsing byte-by-byte.
+            guard let stream = InputStream(fileAtPath: "/dev/stdin") else { throw Error._invalidStdin() }
+            // B.1. Open the stream for usage.
+            assert(stream.streamStatus == .notOpen)
+            stream.open()
+
+            let (encoding, unusedBytes): (String.Encoding, [UInt8])
+            do {
+                // B.2. Check whether the input data has a BOM.
+                let inferred = try String.Encoding.infer(from: stream)
+                // B.3. Select the appropriate encoding depending from the user provided encoding (if any), and the BOM encoding (if any).
+                encoding = try CSVReader.selectEncodingFrom(provided: configuration.encoding, inferred: inferred.encoding)
+                unusedBytes = inferred.unusedBytes
+            } catch let error {
+                if stream.streamStatus != .closed { stream.close() }
+                throw error
```

Comment:
Most of this body is similar to `init` with `Data`. Maybe extract a private method?

Reply:
You  mean `init(input: URL)`? If we expose `InputStream`, we can make `init(input: URL)` a convenience for `init(input: InputStream)`.

Reply:
Sorry, yeah, I think was confused on if you wanted to have `InputStream` public.